lib/puppet/util/pidlock.rb

```diff
@@ -58,11 +69,21 @@ def clear_if_stale
     end
 
     # Ensure the process associated with this pid is our process. If
-    # not, we can unlock the lockfile. For now this is only done on
-    # POSIX and Windows platforms (PUP-9247).
+    # not, we can unlock the lockfile. CLI arguments used for identifying
+    # on POSIX depend on the os and sometimes even version.
     if Puppet.features.posix?
-      procname = Puppet::Util::Execution.execute(["ps", "-p", lock_pid, "-o", "comm="]).strip
-      args     = Puppet::Util::Execution.execute(["ps", "-p", lock_pid, "-o", "args="]).strip
+      ps_argument = ps_argument_for_current_kernel
+
+      begin
+        procname = Puppet::Util::Execution.execute(["ps", ps_argument, lock_pid, "-o", "comm="]).strip
+        args     = Puppet::Util::Execution.execute(["ps", ps_argument, lock_pid, "-o", "args="]).strip
```

Comment:
this could be moved after the block as working argument shall have been already found

Reply:
Agreed. Moved and added a short comment. Might save someone from the headache of following that block of code 🤣 ..